### PR TITLE
Migrate docs to Markdown and update docs

### LINKED ISSUE #77: Add documentation for specifying package version
The readthedocs website should add the following documentation:

Package looks like x.y.z (currently it is `0.8.1`).

The following is when to update each package:
`x`: major update, usually should be done after something is done (ex. progress after a year or a feature is added)

`y`: package change/update

`z`: minor change, refactoring, etc

### LINKED ISSUE #91: Move documentation back to Markdown
We used to have our documentation in Markdown (back when we had the GCS GitHub Wiki, which is now long deleted), and moved it to restructured text to have it on readthedocs. We currently use a tool called Sphinx to let us write documentation that works with the readthedocs website. Sphinx only really works with restructured text, not Markdown.

However, I noticed another tool called MkDocs that lets us write Markdown. I think migrating back to markdown is a good idea. Not sure if this is the best idea, but I think it is easier for newer developers to read markdown.

If so, I can cancel #90, and simply re-write the docs in Markdown (should take 1-2 days at most) with MkDocs. I would like everyone's opinion on this if I should close this issue or continue with the idea...

The following things are what I must do to migrate back to MarkDown:

- [x] figure out how to write notes and warnings in MkDocs Markdown
- [x] setup files to configure MkDocs with readthedocs
- [x] add a [markdown linter](https://github.com/airbnb/javascript/blob/master/linters/.markdownlint.json)

### LINKED ISSUE #95: Add documentation for testing & message updates
See changes to #94

### PR DESCRIPTION
**Pull request by Luis729**
## Why is the change being made?

This change is made because reStructured text could be a more difficult
style of documentation to adapt to (for newer developers) than Markdown.

 ## What has changed to address the problem?

This change removes the Sphinx reStructured text documentation in GCS
and replaces it with a MkDocs Markdown text documentation structure.

This change also adds a linter to the Markdown code.

Everyone contributing to the project will have to follow the
instructions under `docs/contributing/getting-started.md`.

The following are minor changes:
- Ignore files were added to address the `site` folder generated from `mkdocs build` and
removed `docs/_build` from Sphinx.
- Removed dependency to Python 2.
- Specified dependency to Python 3.7 as MkDocs requires it.
- Removed examples. It is hard for a person to develop it. A new way of showing examples will be worked on soon.

There is additional added documentation to address the following:
- How to work with GCS (running, building, and testing)
- Installing MkDocs to see docs locally in `localhost:8000`.
- Adding control messages for manual vehicle control
- Adding geofencing field to start messages

 ## How was this change tested?

This change was tested by checking if the docs looked fine when running
`npm run docs`.

 ## Related documents, URLs, commits
 
Fixes #77
Fixes #91
Fixes #95

### CODE REVIEW

**Comment by Luis729** [comment]
# I suggest the best way to check the changes out yourself is to just run it locally

I havent checked too hard if the links work, so if it breaks let me know

File: docs/communications/messages/base-message.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+## Implementing these fields
+
+!!! warning "Deprecation notice"
+    This section will be deleted once implementation of JSON communication is finished.
+
+  - **type : string**
+      - This field is predefined depending on the type of message.
+
+  - **id : unsigned 32-bit integer**
+      - See [this](../../implementation/#creating-messages-with-proper-id-field) for
```
Comment (ckid199) [inline docs/communications/messages/base-message.md]: All links on this page are broken.

File: docs/communications/messages/gcs-vehicles-messages.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
+# Messages from GCS to vehicles
+
+## Connection acknowledgement message
+
+Sent back to the vehicle that GCS has accepted its request to connect. GCS forwards its time in the
+`time` field for the vehicle to create an offset to and send all future messages to in GCS's time.
+
+See the connect message that initiates this acknowledgement request
+[here](../vehicles-gcs-messages/#connect-message). Also see the implementation for the `time`
```
Comment (ckid199) [inline docs/communications/messages/gcs-vehicles-messages.md]: This link is broken.

File: docs/communications/messages/vehicles-gcs-messages.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
+# Messages from vehicles to GCS
+
+## Connect message
+
+Sent to the GCS to request connection to it. The vehicle will not have information of what GCS's
+time is yet, so any number is sufficient. The GCS will disregard this time.
+
+See the implementation for the ``time`` field for messages, which directly relates to this message
+[here](implementation/#setting-time).
```
Comment (ckid199) [inline docs/communications/messages/vehicles-gcs-messages.md]: Broken link.

File: docs/communications/messages/vehicles-gcs-messages.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  "sid": <unsigned 32-bit integer>,
+  "tid": <unsigned 32-bit integer>,
+  "time": <unsigned 64-bit integer>,
+  "jobsAvailable": <string[]>
+}
+```
+
+  - **jobsAvailable : string[]**
+      - List of jobs that the vehicle is. These jobs describe the tasks the vehicle is capable of
+      performing. See the [list of jobs and tasks][] to see which jobs are valid.
```
Comment (ckid199) [inline docs/communications/messages/vehicles-gcs-messages.md]: Broken link.

File: docs/communications/messages/gcs-vehicles-messages.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+field for messages, which directly relates to this message [here](../../implementation/#setting-time).
```
Comment (ckid199) [inline docs/communications/messages/gcs-vehicles-messages.md]: Broken link.

File: docs/communications/jobs-tasks/jobs-vs-tasks.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+For example, an important part of the project is the ISR Search mission. To perform this mission,
+the GCS look for vehicles that have an `isrSearch` job. By having the `isrSearch` job, the vehicle
+is stating that it is capable of performing all of the following tasks that are related to the job:
+
+  - `takeoff` task
+  - `loiter` task
+  - `isrSearch` task
+  - `land` task
+
+The GCS will assign the vehicles the `isrSearch` job by sending them a [start message][]. By doing
```
Comment (ckid199) [inline docs/communications/jobs-tasks/jobs-vs-tasks.md:14]: I'm assuming [start message] was supposed to link to the start message format page?

File: docs/communications/jobs-tasks/jobs-vs-tasks.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+this, the vehicles know that they will be performing tasks very soon. The start message has a
+`jobType` field, which will be set to `isrSearch`, so that the vehicles know which tasks to expect
+next from the GCS.
+
+To accomplish the ISR Search mission, GCS will assign a task for the vehicle to perform (for
+example, `takeoff` task), and assigns the next after the vehicle completes the first. This goes on
+until the ISR Search mission is completed.
+
+[connect message]: ../../messages/vehicles-gcs-messages/#connect-message
```
Comment (ckid199) [inline docs/communications/jobs-tasks/jobs-vs-tasks.md]: This link (?) doesn't appear at all on the page.

**Review by ckid199 — CHANGES_REQUESTED** [review]
Quickly scanned through the docs for broken links only. Commented on the ones I found.

File: docs/communications/messages/base-message.md (hunk 1 above)
Comment (Luis729) [reply docs/communications/messages/base-message.md]: Fixed

File: docs/communications/messages/vehicles-gcs-messages.md (hunk 3 above)
Comment (Luis729) [reply docs/communications/messages/vehicles-gcs-messages.md]: Fixed

File: docs/communications/messages/vehicles-gcs-messages.md (hunk 4 above)
Comment (Luis729) [reply docs/communications/messages/vehicles-gcs-messages.md]: Fixed

File: docs/communications/messages/gcs-vehicles-messages.md (hunk 5 above)
Comment (Luis729) [reply docs/communications/messages/gcs-vehicles-messages.md]: Fixed

File: docs/communications/jobs-tasks/jobs-vs-tasks.md (hunk 6 above)
Comment (Luis729) [reply docs/communications/jobs-tasks/jobs-vs-tasks.md:14]: Fixed

File: docs/communications/jobs-tasks/jobs-vs-tasks.md (hunk 7 above)
Comment (Luis729) [reply docs/communications/jobs-tasks/jobs-vs-tasks.md]: Fixed, really good catch

Probably will need to add a rule to detect these unused links

**Review by ckid199 — APPROVED** [review]

**Pull request merged**